common/deftype/deftype.cpp

```diff
@@ -1427,9 +1435,29 @@ bool CModifierTypeInfo::equals(const CTypeInfo & _other) const
     return (baseType == other.baseType) && (kind == other.kind) && (extra == other.extra);
 }
 
-
 //===========================================================================
 
+template<typename T>
+T* getOrCreateSingletonAtomic(std::atomic<T*>& singleton)
```

Comment:
style: I don't like the implicit creation of the object based on the type of the parameter - it is very unclear from the caller what is going to be created.  Better would  be to pass in a factory method similar to querySingleton in jmutex.hpp
note: Is there a reason not to use querySingleton()?

Reply:
I didn't realize a function like querySingleton already existed I will use that instead.